server/src/main/java/org/apache/uniffle/server/HealthCheck.java

```diff
@@ -88,15 +88,13 @@ public HealthCheck(
   public void check() {
     for (Checker checker : checkers) {
       if (!checker.checkIsHealthy()) {
-        serverStatus.set(ServerStatus.UNHEALTHY);
+        serverStatus.compareAndSet(ServerStatus.ACTIVE, ServerStatus.UNHEALTHY);
```

Comment:
The `HealthCheck` should only transition from `ACTIVE` to `UNHEALTHY` and back. Otherwise, it could stop the decommissioning phase.